enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,358 @@
+config secret while also respecting the environment variables that would be set by the webhook. Additionally, upstream projects may reject the notion of reading these details from a config secret but that has yet to be seen.
+
+## Design Details
+
+### Open Questions [optional]
+
+- From where should CCO source the mutating admission webhook for deployment? In order to generate our own build of the image backing the webhook we would have to fork [Azure/azure-workload-identity](https://github.com/Azure/azure-workload-identity)([dockerfile](https://github.com/Azure/azure-workload-identity/blob/main/docker/webhook.Dockerfile)).
+
+### Test Plan
+
+An e2e test job will be created similar to the [e2e-gcp-manual-oidc](https://github.com/openshift/release/pull/22552) that,
+- Extracts `CredentialsRequests` from the release image.
+- Processes `CredentialsRequests` with `ccoctl` to generate secret and `Authentication` configuration manifests.
+- Moves the generated manifests into the manifests directory used for install.
+- Runs the normal e2e suite against the resultant cluster.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+As clusters are upgraded, new permissions may be required or extended (in the case of future role granularity work) and users must evaluate those changes at the upgrade boundary similarly to [upgrading an STS cluster in manual mode](https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-manual.html#manual-mode-sts-blurb).
+
+### Version Skew Strategy
+
+None.
+
+### Operational Aspects of API Extensions
+
+None.
+
+#### Failure Modes
+
+None.
+
+#### Support Procedures
+
+- How to detect that operator credentials are incorrect / insufficient?
+  - ClusterOperators will be degraded when credentials are not present / insufficient.
```

Comment:
There will be a test for this?  I don't think such things are necessarily automatic depending on the shape of the operator.

Reply:
True, I would assume that operators are aware that they don't have credentials today but aren't aware that their credentials are insufficient. CCO will be disabled since clusters utilizing Azure workload identity would be in manual mode so there would be no ongoing reconciliation ensuring that the credentials are sufficient.

Reply:
Additionally, ensuring that the credentials are sufficient requires privileged credentials which don't exist on clusters in `manual` mode.

Other platforms like AWS and GCP have this validation in `mint` and `passthrough` modes but `manual` mode exists so that privileged credentials do not live on the cluster.